### ISSUE TITLE
STEP 1 error again

### ISSUE THREAD

**Issue by leehwmj** [jtamames/SqueezeMeta#336]
Hello, I'm using it well thanks to the developer.

I am inquiring using the issue because there has been a new problem recently.

When executed, the following error message occurs.
`ERROR: input file for -fasta is in UNKNOWN format not in FASTA format.`

↓ full error 
```
$ SqueezeMeta.pl -m coassembly -p antarctic_ocean_dupli -s antarctic_ocean_dupli.samples -f ./seq/ --cleaning -t 60

SqueezeMeta v1.4.0, May 2021 - (c) J. Tamames, F. Puente-Sánchez CNB-CSIC, Madrid, SPAIN

Please cite: Tamames & Puente-Sanchez, Frontiers in Microbiology 9, 3349 (2019). doi: https://doi.org/10.3389/fmicb.2018.03349

Run started Tue Jul 27 11:04:11 2021 in coassembly mode
Now creating directories
Reading configuration from /analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/SqueezeMeta_conf.pl
Reading samples from /analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/00.antarctic_ocean_dupli.samples
mv: '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-1_TTAGGC_L001_R1_001.fq.gz' and '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-1_TTAGGC_L001_R1_001.fq.gz' are the same file
mv: '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-1_TTAGGC_L001_R2_001.fq.gz' and '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-1_TTAGGC_L001_R2_001.fq.gz' are the same file
  Running trimmomatic (Bolger et al 2014, Bioinformatics 30(15):2114-20) for quality filtering
  Parameters: LEADING:8 TRAILING:8 SLIDINGWINDOW:10:15 MINLEN:30
mv: '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-1_TTAGGC_L002_R1_001.fq.gz' and '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-1_TTAGGC_L002_R1_001.fq.gz' are the same file
mv: '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-1_TTAGGC_L002_R2_001.fq.gz' and '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-1_TTAGGC_L002_R2_001.fq.gz' are the same file
  Running trimmomatic (Bolger et al 2014, Bioinformatics 30(15):2114-20) for quality filtering
  Parameters: LEADING:8 TRAILING:8 SLIDINGWINDOW:10:15 MINLEN:30
mv: '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-2_ACTTGA_L001_R1_001.fq.gz' and '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-2_ACTTGA_L001_R1_001.fq.gz' are the same file
mv: '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-2_ACTTGA_L001_R2_001.fq.gz' and '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-2_ACTTGA_L001_R2_001.fq.gz' are the same file
  Running trimmomatic (Bolger et al 2014, Bioinformatics 30(15):2114-20) for quality filtering
  Parameters: LEADING:8 TRAILING:8 SLIDINGWINDOW:10:15 MINLEN:30
mv: '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-2_ACTTGA_L002_R1_001.fq.gz' and '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-2_ACTTGA_L002_R1_001.fq.gz' are the same file
mv: '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-2_ACTTGA_L002_R2_001.fq.gz' and '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/16-2_ACTTGA_L002_R2_001.fq.gz' are the same file
  Running trimmomatic (Bolger et al 2014, Bioinformatics 30(15):2114-20) for quality filtering
  Parameters: LEADING:8 TRAILING:8 SLIDINGWINDOW:10:15 MINLEN:30
mv: '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/25_GGCTAC_L001_R1_001.fq.gz' and '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/25_GGCTAC_L001_R1_001.fq.gz' are the same file
mv: '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/25_GGCTAC_L001_R2_001.fq.gz' and '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/25_GGCTAC_L001_R2_001.fq.gz' are the same file
  Running trimmomatic (Bolger et al 2014, Bioinformatics 30(15):2114-20) for quality filtering
  Parameters: LEADING:8 TRAILING:8 SLIDINGWINDOW:10:15 MINLEN:30
mv: '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/25_GGCTAC_L002_R1_001.fq.gz' and '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/25_GGCTAC_L002_R1_001.fq.gz' are the same file
mv: '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/25_GGCTAC_L002_R2_001.fq.gz' and '/analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/raw_fastq/25_GGCTAC_L002_R2_001.fq.gz' are the same file
  Running trimmomatic (Bolger et al 2014, Bioinformatics 30(15):2114-20) for quality filtering
  Parameters: LEADING:8 TRAILING:8 SLIDINGWINDOW:10:15 MINLEN:30
6 samples found: 16depth-2-L2 25depth-L1 16depth-1-L2 16depth-1-L1 16depth-2-L1 25depth-L2

Now merging files
[7 seconds]: STEP1 -> RUNNING CO-ASSEMBLY: 01.run_assembly.pl (megahit)
  Running assembly with megahit
  Running prinseq (Schmieder et al 2011, Bioinformatics 27(6):863-4) for selecting contigs longer than 200 

ERROR: input file for -fasta is in UNKNOWN format not in FASTA format.

Try 'perl prinseq-lite.pl -h' for more information.
Exit program.
Error running command:    /home/hwlee/anaconda3/envs/SqueezeMeta/SqueezeMeta/bin/prinseq-lite.pl -fasta /analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/data/megahit/final.contigs.fa -min_len 200 -out_good /analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/results/prinseq; mv /analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/results/prinseq.fasta /analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/results/01.antarctic_ocean_dupli.fasta > /dev/null 2>&1 at /home/hwlee/anaconda3/envs/SqueezeMeta/SqueezeMeta/scripts/01.run_assembly.pl line 130.
Stopping in STEP1 -> 01.run_assembly.pl. Program finished abnormally

  If you don't know what went wrong or want further advice, please look for similar issues in https://github.com/jtamames/SqueezeMeta/issues
  Feel free to open a new issue if you don't find the answer there. Please add a brief description of the problem and upload the /analysis/hwlee/antartic_ocean/antarctic_ocean_dupli/syslog file (zip it first)
Died at /home/hwlee/anaconda3/envs/SqueezeMeta/bin/SqueezeMeta.pl line 1367.
```

I ran `test_install.pl` because I remember experiencing and resolving similar errors before.
However, all checks were successful.
```
$ test_install.pl

Checking the OS
	linux OK

Checking that tree is installed
	tree --help OK

Checking that ruby is installed
	ruby -h OK

Checking that java is installed
	java -h OK

Checking that all the required perl libraries are available in this environment
	perl -e 'use Term::ANSIColor' OK
	perl -e 'use DBI' OK
	perl -e 'use DBD::SQLite::Constants' OK
	perl -e 'use Time::Seconds' OK
	perl -e 'use Tie::IxHash' OK
	perl -e 'use Linux::MemInfo' OK
	perl -e 'use Getopt::Long' OK
	perl -e 'use File::Basename' OK
	perl -e 'use DBD::SQLite' OK
	perl -e 'use Data::Dumper' OK
	perl -e 'use Cwd' OK
	perl -e 'use XML::LibXML' OK
	perl -e 'use XML::Parser' OK
	perl -e 'use Term::ANSIColor' OK

Checking that all the required python libraries are available in this environment
	python3 -h OK
	python3 -c 'import numpy' OK
	python3 -c 'import scipy' OK
	python3 -c 'import matplotlib' OK
	python3 -c 'import dendropy' OK
	python3 -c 'import pysam' OK
	python3 -c 'import pandas' OK
	python3 -c 'import cython' OK
	python3 -c 'import future' OK

Checking that all the required R libraries are available in this environment
	R -h OK
	R -e 'library(doMC)' OK
	R -e 'library(ggplot2)' OK
	R -e 'library(data.table)' OK
	R -e 'library(reshape2)' OK
	R -e 'library(pathview)' OK
	R -e 'library(DASTool)' OK
	R -e 'library(SQMtools)' OK

Checking that SqueezeMeta is properly configured... checking database in /analysis/database/SqueezeMetaDB/db
	nr.db OK
	CheckM manifest OK
	LCA_tax DB OK

All checks successful
```

The contents of the Samples file are as follows.
```
$ cat antarctic_ocean_dupli.samples 
16depth-1-L1	16-1_TTAGGC_L001_R1_001.fq.gz	pair1
16depth-1-L1	16-1_TTAGGC_L001_R2_001.fq.gz	pair2
16depth-1-L2	16-1_TTAGGC_L002_R1_001.fq.gz	pair1
16depth-1-L2	16-1_TTAGGC_L002_R2_001.fq.gz	pair2
16depth-2-L1	16-2_ACTTGA_L001_R1_001.fq.gz	pair1
16depth-2-L1	16-2_ACTTGA_L001_R2_001.fq.gz	pair2
16depth-2-L2	16-2_ACTTGA_L002_R1_001.fq.gz	pair1
16depth-2-L2	16-2_ACTTGA_L002_R2_001.fq.gz	pair2
25depth-L1	25_GGCTAC_L001_R1_001.fq.gz	pair1
25depth-L1	25_GGCTAC_L001_R2_001.fq.gz	pair2
25depth-L2	25_GGCTAC_L002_R1_001.fq.gz	pair1
25depth-L2	25_GGCTAC_L002_R2_001.fq.gz	pair2
```

and add syslog file
[syslog.txt](https://github.com/jtamames/SqueezeMeta/files/6882371/syslog.txt)

I tried to resolve it in the following ways, but we didn't get the desired result:
https://github.com/jtamames/SqueezeMeta/issues/258#issuecomment-797595449

Please give me good advice.
Thanks

**Comment by leehwmj** [comment]
It's a fact that I found out during the analysis.

When --cleaning was deleted and executed, it was analyzed normally.
I wonder how to proceed with the analysis with --cleaning.

**Comment by jtamames** [comment]
Hello
I assume you installed SqueezeMeta via conda. If so, locate you conda env via the command `conda info --envs`.  There, edit the script SqueezeMeta.pl that you will find in the SqueezeMeta/scripts directory in that conda env. Just before line 434, reading:

`my $tcommand="mv $par1name $orig1; mv $par2name $orig2";`

add this:

`
$orig1=~s/\.fq/\.original.fastq/;
$orig2=~s/\.fq/\.original.fastq/;
`

And run it again. Let me know if it works.
Best,
Javier

**Comment by leehwmj** [comment]
Thanks jtamames!

I add the code to the script.
I will run it then when I get success, clese this issue.

**Comment by leehwmj** [comment]
ah. In addition, I have another question.

After analyzing reads using sqm_reads.pl, and tables were created using sqmreads2tables.py.

When summarized as follows, each KEGG names is given numbers.
![image](https://user-images.githubusercontent.com/22087964/127263241-ab2655c7-8543-48e9-836a-7e15e92d1651.png)

And I have checked the explanation as below in the manual.
![image](https://user-images.githubusercontent.com/22087964/127263153-d33065c9-9171-4647-8b7c-8a7450663819.png)

At this time, I'm curious about the definition of "abundunce" on the table.
How it calculated?

It's not raw reads count, it's not contigs number.
Found "raw abundance" in the published manual (SqueezeMeta_manual_v1.4.0) (page 41), but not enough to describe raw abundances in sqmreads2tables.py result "<project_name>.<classification>.abunds.tsv".
